googletest/feature_support_test.cpp

```diff
@@ -0,0 +1,1313 @@
+}
+
+// 0: D3D12_OPTIONS
+// Arbitarily set support status for caps and see if the returned results are correct
+TEST_F(FeatureSupportTest, D3D12Options) {
+    device->m_DoublePrecisionFloatShaderOps = true;
+    device->m_OutputMergerLogicOp = true;
+    device->m_ShaderMinPrecisionSupport10Bit = D3D12_SHADER_MIN_PRECISION_SUPPORT_10_BIT;
+    device->m_ShaderMinPrecisionSupport16Bit = D3D12_SHADER_MIN_PRECISION_SUPPORT_16_BIT;
+    device->m_TiledResourcesTier = D3D12_TILED_RESOURCES_TIER_3;
+    device->m_ResourceBindingTier = D3D12_RESOURCE_BINDING_TIER_3;
+    device->m_PSSpecifiedStencilRefSupported = true;
+    device->m_ConservativeRasterizationTier = D3D12_CONSERVATIVE_RASTERIZATION_TIER_2;
+    device->m_MaxGPUVirtualAddressBitsPerResource = 10;
+    device->m_ResourceHeapTier = D3D12_RESOURCE_HEAP_TIER_2;
+
+    INIT_FEATURES();
+    EXPECT_TRUE(features.DoublePrecisionFloatShaderOps());
+    EXPECT_TRUE(features.OutputMergerLogicOp());
+    EXPECT_EQ(features.MinPrecisionSupport(), D3D12_SHADER_MIN_PRECISION_SUPPORT_10_BIT | D3D12_SHADER_MIN_PRECISION_SUPPORT_16_BIT);
+    EXPECT_EQ(features.TiledResourcesTier(), D3D12_TILED_RESOURCES_TIER_3);
+    EXPECT_EQ(features.ResourceBindingTier(), D3D12_RESOURCE_BINDING_TIER_3);
+    EXPECT_TRUE(features.PSSpecifiedStencilRefSupported());
+    EXPECT_FALSE(features.TypedUAVLoadAdditionalFormats());
+    EXPECT_FALSE(features.ROVsSupported());
+    EXPECT_EQ(features.ConservativeRasterizationTier(), D3D12_CONSERVATIVE_RASTERIZATION_TIER_2);
+    EXPECT_EQ(features.MaxGPUVirtualAddressBitsPerResource(), 10);
+    EXPECT_FALSE(features.StandardSwizzle64KBSupported());
+    EXPECT_EQ(features.CrossNodeSharingTier(), D3D12_CROSS_NODE_SHARING_TIER_NOT_SUPPORTED);
+    EXPECT_FALSE(features.CrossAdapterRowMajorTextureSupported());
+    EXPECT_FALSE(features.VPAndRTArrayIndexFromAnyShaderFeedingRasterizerSupportedWithoutGSEmulation());
+    EXPECT_EQ(features.ResourceHeapTier(), D3D12_RESOURCE_HEAP_TIER_2);
+}
+
+// 1: Architecture & 16: Architecture1
+// Test where architecture1 is available
+TEST_F(FeatureSupportTest, Architecture1Available)
+{
+    device->m_Architecture1Available = true;
+    device->m_TileBasedRenderer[0] = false;
+    device->m_UMA[0] = true;
+    device->m_CacheCoherentUMA[0] = true;
+    device->m_IsolatedMMU[0] = true;
+
+    INIT_FEATURES();
+    EXPECT_FALSE(features.TileBasedRenderer(0));
+    EXPECT_TRUE(features.UMA(0));
+    EXPECT_TRUE(features.CacheCoherentUMA(0));
+    EXPECT_TRUE(features.IsolatedMMU(0));
+}
+
+// Test where architecture1 is not available
+TEST_F(FeatureSupportTest, Architecture1Unavailable)
+{
+    device->m_Architecture1Available = false; // Architecture1 not available
+    device->m_TileBasedRenderer[0] = false;
+    device->m_UMA[0] = true;
+    device->m_CacheCoherentUMA[0] = false;
+    device->m_IsolatedMMU[0] = true; // Notice that if architecture1 is not available, IslatedMMU cap should not be present
```

Comment:
If Architecture1 is unavailable, the API will use results from the old Architecture for the corresponding fields. `IsolatedMMU` is not included in the old Architecture struct, so it should return false in this case.